File: backend/src/main/java/com/kh/backend/domain/user/model/vo/Dog.java
package com.kh.backend.domain.user.model.vo;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Dog {

	private int dogNo;
	private int userNo;
	private char isMain; // DB에서는 char
	private String dogName;
	private String breed; // "견종 직접 작성 가능	(디폴트 모름)"
	private char gender; // "성별 선택 M - 수컷 F - 암컷 N - 모름"
	private String birthday; // 일단 String
	private String note;
}

File: service-frontend/src/type/dog.ts
export interface ImgDog{
    dogNo : number,
    originName : string,
    changeName : string
}

export const initImgDog:ImgDog = {
    dogNo : 0,
    originName : '',
    changeName : ''
}

export interface Dog{
    dogNo : number,
    userNo : number,
    isMain : string,
    dogName : string,
    breed : string,
    gender : string,
    birthday : string,
    note : string,
    imgDog : ImgDog
}

export const initDog:Dog = {
    dogNo : 0,
    userNo : 0,
    isMain : '',
    dogName : '',
    breed : '',
    gender : '',
    birthday : '',
    note : '',
    imgDog : initImgDog
} as const;

export const initDogList:Dog[] = [];